lib/internal/vm/module.js

```diff
@@ -348,6 +363,16 @@ class SourceTextModule extends Module {
     }
     return super.error;
   }
+
+  createCachedData() {
+    const { status } = this;
+    if (status === 'evaluating' ||
+        status === 'evaluated' ||
+        status === 'errored') {
```

Comment:
Would it maybe make sense to use a whitelist instead of a blacklist?

Reply:
The condition is "the module is not evaluated", so I think specifically checking for evaluated makes more sense.